saleor/graphql/checkout/mutations.py

```diff
@@ -861,10 +890,17 @@ def perform_mutation(cls, _root, info, checkout_id, store_source, **data):
             # remove checkout after order is successfully paid
             checkout.delete()
 
-            # return the success response with the newly created order data
-            return CheckoutComplete(order=order, confirmation_needed=False)
+        else:
+            # If gateway returns information that additional steps are required we need
+            # to inform the frontend and pass all required data
+            confirmation_needed = True
+            confirmation_data = txn.action_required_data
 
-        return CheckoutComplete(order=None, confirmation_needed=True)
+        return CheckoutComplete(
+            order=order,
+            confirmation_needed=confirmation_needed,
+            confirmation_data=confirmation_data,
```

Comment:
```suggestion
            confirmation_needed=True if txn.action_required else False,
            confirmation_data={} if txn.action_required else txn.action_required_data,
```
Maybe something like this? Now we set two variables above and then modify them or not to finally return them, that way it's in one place and `else` can be removed completely